docs/source/federated-types/federated-directives.mdx

```diff
@@ -0,0 +1,302 @@
+---
+title: Federation-specific GraphQL directives
+---
+
+Apollo Federation defines a collection of **directives** that you use in your subgraph schemas to enable certain features.
+
+## Importing directives
+
+To use federated directives in a Federation 2 subgraph schema, apply the `@link` directive with the following format to the `schema` type:
+
+```graphql
+extend schema
+  @link(url: "https://specs.apollo.dev/federation/v2.0",
+        import: ["@key", "@shareable"])
+```
+
+You can apply this directive to your existing `schema` declaration if you have one, or to a new `extend schema` declaration (as shown above).
+
+Modify the `import` array to include whichever federated directives your subgraph schema uses. The example above imports the [`@key`](#key) and [`@shareable`](#shareable) directives (which are used most commonly).
+
+> Make sure to include the `@` in each directive name!
+
+### Renaming directives
+
+If a federated directive's name matches one of your own custom directives, you can rename the federated directive with the following syntax:
+
+```graphql
+extend schema
+  @link(url: "https://specs.apollo.dev/federation/v2.0",
+        import: [{ name: "@key", as: "@primaryKey"}, "@shareable"])
+```
+
+This example subgraph schema uses `@primaryKey` for the federated directive usually named [`@key`](#key).
+
+## Managing entities
+
+### `@key`
+
+Designates an object type as an [entity](../entities/).
+
+**Valid locations:** Object type definitions
+
+```graphql {1}
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+#### Arguments
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+
+<tr class="required">
+<td>
+
+##### `fields`
+
+`String`
```

Comment:
> My favorite solution would be to host the actual (GraphQL) specification at https://specs.apollo.dev/federation/v2.0

It'll be hosted there eventually, it's just not ready yet (but we'll certainly try to have "something" available there for GA or immediately afterwards).

> I get that this might be too much detail in a high level doc but it's also confusing for someone reading another definition from somewhere else.

I do both agree with the risk of confusion and worry that using `FieldSet` might unnecessarily distract here.

At the same time, while I do think we should link to the spec when it's available, I like that we describe each argument with a minimum of details here and that the page stand on its own. I wouldn't want that page to be reduced to a list of examples.

A possible cope-out could be to have this doc describe the "values" the argument accept, but without mentioning the type per-se. That is, instead of writing "`String`" here (and other similar places of this doc), we could write "a string value".

Reply:
Docs are hard 😅 . I think I'd just leave out the "type" and let the description speak for itself but having something like "accepted values" works too 👍

Reply:
I've cut the `String` type from each table and refer to each value as "A GraphQL field selection set (provided as a string) blah blah"

Is that roughly what we're going for?

Reply:
LGTM! Thanks!

Reply:
Same, looks great to me.